src/main/java/org/opentripplanner/routing/algorithm/filterchain/ItineraryFilterChainBuilder.java

```diff
@@ -71,15 +91,6 @@ public ItineraryFilter build() {
             }
         }
 
-        // Sort list on {@code groupByP} in ascending order to keep as many of the elements in the
-        // groups where the grouping parameter is relaxed as possible.
-        List<GroupBySimilarity> groupBy = parameters.groupBySimilarity().stream()
-            .sorted(Comparator.comparingDouble(o -> o.groupByP))
-            .collect(Collectors.toList());
-
-        for (GroupBySimilarity it : groupBy) {
-            filters.add(new GroupBySimilarLegsFilter(it.groupByP, it.approximateMinLimit));
-        }
 
         // Sort itineraries
         filters.add(new OtpDefaultSortOrder(parameters.arriveBy()));
```

Comment:
My idea was that we should filter out all itineraries which are sorted to be after the first one which does not depart within the search window. That way we are guaranteed to return no worse itineraries than that one, and requiring the new search to include the first one removed.

Reply:
Discussed ver Slack, no need to do this now.